internal/output/output.go

```diff
@@ -2,20 +2,44 @@ package output
 
 import (
 	"encoding/json"
-	"os"
 
 	"github.com/fatih/color"
 	"github.com/pingidentity/pingctl/internal/logger"
+	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 )
 
 var (
-	green = color.New(color.FgGreen).PrintfFunc()
+	cyan   = color.New(color.FgCyan).SprintfFunc()
+	green  = color.New(color.FgGreen).SprintfFunc()
+	red    = color.New(color.FgRed).SprintfFunc()
+	white  = color.New(color.FgWhite).SprintfFunc()
+	yellow = color.New(color.FgYellow).SprintfFunc()
 )
 
-func Format(fields map[string]interface{}) {
+type CommandOutputResult string
+
+type CommandOutput struct {
+	Fields  map[string]interface{}
+	Message string
+	Result  CommandOutputResult
+}
+
+const (
+	ENUMCOMMANDOUTPUTRESULT_NIL           CommandOutputResult = ""
+	ENUMCOMMANDOUTPUTRESULT_SUCCESS       CommandOutputResult = "Success"
+	ENUMCOMMANDOUTPUTRESULT_NOACTION_OK   CommandOutputResult = "No Action (OK)"
+	ENUMCOMMANDOUTPUTRESULT_NOACTION_WARN CommandOutputResult = "No Action (Warning)"
+	ENUMCOMMANDOUTPUTRESULT_FAILURE       CommandOutputResult = "Failure"
+)
+
+func Format(cmd *cobra.Command, output CommandOutput) {
```

Comment:
The difference is developer communication versus user communication. I believe most logs should be for debug, development, or triage purposes. Anything being sent to the customer should hopefully be applied through this output formatter. I think this might mean in some cases, we will need to double our error messaging, one to logger and the other to stdout for the customer to have feedback on a failed command call. Thoughts?

Reply:
We could update this method to add anything printed to the user to the logger at a lower level as well.

It would be nice if there was a simpler interface if the tool just wants to print a simple progress message ("Waiting for response..." for example). This method is a little particular about what it expects.